tests/bor/bor_reorg_test.go

```diff
@@ -276,3 +278,155 @@ func TestValidatorWentOffline(t *testing.T) {
 	assert.Equal(t, authorVal1, nodes[0].AccountManager().Accounts()[0])
 
 }
+
+func TestForkWithBlockTime(t *testing.T) {
+
+	cases := []struct {
+		sprint        uint64
+		blockTime     map[string]uint64
+		change        uint64
+		producerDelay uint64
+		forkExpected  bool
+	}{
+		{
+			sprint: 128,
+			blockTime: map[string]uint64{
+				"0":   5,
+				"128": 2,
+				"256": 8,
+			},
+			change:        2,
+			producerDelay: 8,
+			forkExpected:  false,
+		},
+		{
+			sprint: 64,
+			blockTime: map[string]uint64{
+				"0":  5,
+				"64": 2,
+			},
+			change:        1,
+			producerDelay: 5,
+			forkExpected:  false,
+		},
+		{
+			sprint: 16,
+			blockTime: map[string]uint64{
+				"0":  2,
+				"64": 5,
+			},
+			change:        4,
+			producerDelay: 4,
+			forkExpected:  true,
+		},
+	}
+	log.Root().SetHandler(log.LvlFilterHandler(log.LvlInfo, log.StreamHandler(os.Stderr, log.TerminalFormat(true))))
+	fdlimit.Raise(2048)
+
+	// Generate a batch of accounts to seal and fund with
+	faucets := make([]*ecdsa.PrivateKey, 128)
+	for i := 0; i < len(faucets); i++ {
+		faucets[i], _ = crypto.GenerateKey()
+	}
+
+	// Create an Ethash network based off of the Ropsten config
+	genesis := initGenesis(t, faucets)
+
+	for _, test := range cases {
+		genesis.Config.Bor.Sprint = test.sprint
+		genesis.Config.Bor.Period = test.blockTime
+		genesis.Config.Bor.BackupMultiplier = test.blockTime
+		genesis.Config.Bor.ProducerDelay = test.producerDelay
+
+		var (
+			stacks []*node.Node
+			nodes  []*eth.Ethereum
+			enodes []*enode.Node
+		)
+
+		for i := 0; i < 2; i++ {
+			// Start the node and wait until it's up
+			stack, ethBackend, err := initMiner(genesis, keys[i])
+			if err != nil {
+				t.Fatal("Error occured while initialising miner", "error", err)
+			}
+			defer stack.Close()
+
+			for stack.Server().NodeInfo().Ports.Listener == 0 {
+				time.Sleep(250 * time.Millisecond)
+			}
+			// Connect the node to all the previous ones
+			for _, n := range enodes {
+				stack.Server().AddPeer(n)
+			}
+			// Start tracking the node and its enode
+			stacks = append(stacks, stack)
+			nodes = append(nodes, ethBackend)
+			enodes = append(enodes, stack.Server().Self())
+		}
+
+		// Iterate over all the nodes and start mining
+		time.Sleep(3 * time.Second)
```

Comment:
Why do we need sleep before?

Reply:
I believe we wait for the setup to be completed.

Reply:
Would it work without this sleep? If not could we find and use some kind of event that signals successful initialization?